### PR TITLE
Publish To Gradle Plugin Portal

### LINKED ISSUE #17: Publish the plugin to the Gradle plugin portal
We should publish this plugin to the Gradle plugin portal as it would allow for more publicity and cleaner build scripts for our users. Documentation on publishing to the plugin portal can be found [here](https://plugins.gradle.org/docs/submit). I am more than happy to take this task on myself but i wanted to make sure you guys are OK with the idea first.

### PR DESCRIPTION
**Pull request by 06needhamt**
Plugin is now live at https://plugins.gradle.org/plugin/me.rexim.morganey.gradle

Close #17

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -50,3 +50,24 @@ dependencies {
...
+// Unless overridden in the pluginBundle config DSL, the project version will
+// be used as your plugin version when publishing
+version = "0.1"
+group = "me.rexim.morganey.gradle"
+
+// The configuration example below shows the minimum required properties
+// configured to publish your plugin to the plugin portal
+pluginBundle {
+    website = 'http://www.gradle.org/'
+    vcsUrl = 'https://github.com/gradle/gradle'
```
Comment (ForNeVeR) [inline build.gradle]: Isn't that supposed to be _our_ website and _our_ git repo?

File: plugin-test/build.gradle [hunk 2]
```diff
@@ -15,14 +15,16 @@ sourceCompatibility = 1.8
...
     repositories {
-        flatDir dirs: "../jars/repo/me/rexim/morganey/gradle/morganey-gradle-plugin/unspecified/"
-        mavenCentral()
+        maven {
+            url "https://plugins.gradle.org/m2/"
+        }
     }
-
     dependencies {
-        classpath 'me.rexim.morganey.gradle:morganey-gradle-plugin:unspecified'
```
Comment (ForNeVeR) [inline plugin-test/build.gradle:23]: I can't see the reason of changing our package name (prepending this `gradle.plugin` there). Could you please elaborate?

**Review by ForNeVeR — CHANGES_REQUESTED** [review]

**Comment by ForNeVeR** [comment]
@06needhamt please don't publish anything until my next announcement. I'll try to take measures to change the plugin id and will see how to share the project on the plugin portal.

**Comment by 06needhamt** [comment]
@ForNeVeR Thanks for the review i have addressed your changes.

**Comment by ForNeVeR** [comment]
Okay, I've taken a look at the so-called "new plugin system" and made some changes in our project naming.
1. Project group now is just `me.rexim.morganey` (instead of `me.rexim.morganey.gradle`) because of (2).
2. Project now _actually_ published as `gradle.plugins.me.rexim.morganey`, and that's the reason I've decided to drop that additional `gradle` there. The `gradle.plugins` group is softly enforced by Gradle team themselves, because of
   
   > As publishing to a custom group requires manual approval by the Gradle
   > team for security reasons, we recommend not overriding the group ID unless
   > you have an existing group ID that you wish to keep. If not overridden,
   > plugins will be published automatically without a manual approval process.
   
   as they state [here](https://plugins.gradle.org/docs/publish-plugin) (see _"Full Example"_ tab).
3. Plugin usage is simplified through usage of the "new plugin system": just drop a little `plugins` block, no reason to keep these `buildscript` monstrosities.

**Comment by 06needhamt** [comment]
@ForNeVeR Cool will change this now

**Comment by ForNeVeR** [comment]
@rexim @06needhamt please review my latest changes and the previous comment.

**Review by ForNeVeR — APPROVED** [review]

**Comment by ForNeVeR** [comment]
@06needhamt no need to change anything; I've already done that and pushed to your branch. The new plugin version is published under the name I suggested.

**Comment by 06needhamt** [comment]
@ForNeVeR LGTM is there any way to remove the old one from the plugin portal?

**Comment by ForNeVeR** [comment]
@06needhamt yep I think that we need to remove it, but I can't see any way of doing that except contacting the support. You could contact them directly, or let me contact them (I could add you to cc list when writing them anything).

**Comment by 06needhamt** [comment]
@ForNeVeR OK Should I contact them as it my account what do you think

**Review by rexim — APPROVED** [review]

**Comment by ForNeVeR** [comment]
@06needhamt yep.

**Comment by 06needhamt** [comment]
do you have the support address i can only find info@gradle.com

**Comment by ForNeVeR** [comment]
@06needhamt the documentation suggests using [this contact form](https://plugins.gradle.org/docs/get-help).

**Comment by 06needhamt** [comment]
nvm you can just post on the forum

**Comment by ForNeVeR** [comment]
@06needhamt please keep me informed, post the links to the forum etc. if you need help or approval from the Morganey team or something.

**Comment by ForNeVeR** [comment]
Thanks everyone for your review and contributions, we've published our plugin! 🙌

**Comment by 06needhamt** [comment]
Please see forum post [here](https://discuss.gradle.org/t/need-to-remove-this-plugin-me-rexim-morganey-gradle/20136) in regard to removing our old release

**Comment by 06needhamt** [comment]
The old release has been removed 👍

**Pull request merged**